src/xo-app/settings/acls/index.js

```diff
@@ -0,0 +1,216 @@
+import _ from 'messages'
+import ActionButton from 'action-button'
+import { connectStore } from 'utils'
+import { createGetObjectsOfType, createSelector } from 'selectors'
+import find from 'lodash/find'
+import forEach from 'lodash/forEach'
+import isEmpty from 'lodash/isEmpty'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import { error } from 'notification'
+import { GenericSelect, SelectSubject } from 'select-objects'
+import assign from 'lodash/assign'
+import keyBy from 'lodash/keyBy'
+import ObjectName from 'object-name'
+import orderBy from 'lodash/orderBy'
+
+import {
+  addAcl,
+  removeAcl,
+  subscribeAcls,
+  subscribeGroups,
+  subscribeRoles,
+  subscribeUsers
+} from 'xo'
+
+@connectStore(() => {
+  const getPools = createGetObjectsOfType('pool')
+  const getHosts = createGetObjectsOfType('host')
+  const getVms = createGetObjectsOfType('VM')
+  const getSrs = createGetObjectsOfType('SR')
+  const getNetworks = createGetObjectsOfType('network')
+
+  const getHighLevelObjects = createSelector(
+    getPools,
+    getHosts,
+    getVms,
+    getSrs,
+    getNetworks,
+    (pools, hosts, vms, srs, networks) => assign(pools, hosts, vms, srs, networks)
+  )
+
+  return (state, props) => ({objects: getHighLevelObjects(state, props)})
+})
+class SelectObject extends GenericSelect {
+  constructor (props) {
+    super(props)
+    this._placeholder = _('selectObjects')
+  }
+
+  _computeOptions (props) {
+    return map(props.objects, object => ({
+      value: object.id,
+      label: object.name_label || object.name || object.id,
+      type: object.type && object.type.toLowerCase()
+    }))
+  }
+}
+
+class SelectRole extends GenericSelect {
```

Comment:
This select component "glitches", it sometimes displays "No result found" and it seems slow to render when the subscription refreshes. Nothing special to be seen on server side when role.getAll is called...